JCLOUDS-1116: Delegate endpoint should be preferred to the caller's one

nacx opened the pull request:
https://issues.apache.org/jira/browse/JCLOUDS-1116

When computing the endpoint for delegate apis, if both the caller and the callee define the endpoint, then the one in the callee should take precedence.

andrewgaul commented on core/src/test/java/org/jclouds/rest/internal/RestAnnotationProcessorTest.java:
```diff
@@ -411,6 +418,46 @@ public HttpResponse invoke(HttpCommand command) {
+
+      try {
+         child.getInstance(Callee.class);
+         fail("Callee shouldn't be bound yet");
```
Might want to use assertj `failBecauseExceptionWasNotThrown`.

nacx replied:
Done!

andrewgaul commented:
:+1:

zack-shoylev commented on core/src/main/java/org/jclouds/rest/internal/RestAnnotationProcessor.java:
```diff
@@ -198,14 +198,19 @@ public GeneratedHttpRequest apply(Invocation invocation) {
-            endpoint = findEndpoint(invocation);
       } else {
-         endpoint = findEndpoint(invocation);
+         endpoint = getEndpointFor(invocation);
```
Document the new logic a little bit for future reference?

nacx commented:
Squashed, rebased, added comment and merged to master as [040df11d](http://git-wip-us.apache.org/repos/asf/jclouds/commit/040df11d). Thanks for the review!
